c10/util/strides.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include <c10/util/ArrayRef.h>
+
+namespace c10 {
+
+// Computes the contiguous strides of a tensor, given its sizes.
+static inline std::vector<typename IntArrayRef::value_type> contiguous_strides(
```

Comment:
There were a couple of reasons for that:
- `DimVector` is defined in `ATen/core/DimVector.h`. Wasn't sure I could move it to `c10`
- I could use `SmallVector` directly, but I don't think that would be good (since I would have to specify the type parameters myself)
- I didn't think it would make that much of a difference -- only a few places call `contiguous_strides`, and they only call it once or twice

Reply:
As `DimVector` is just a `using...`, probably returning a `SmallVector<typename IntArrayRef::value_type>` works just fine. This should be correct as this is the internal type that the arrayrefs of `sizes()` and `strides()` return.

That being said, yeah, it may not matter too much.

Now my question is, why is `DimVector` not in c10 @ezyang ?

Reply:
It can easily go in c10, feel free to move it. Looking over the history it looks like SmallVector got moved to c10 in 2019 but we forgot to move this file too.

Returning a DimVector is better, it avoids heap allocation in the common case.

Reply:
Implemented on #77045